src/lib/draw/polygon/polygon_tool.ts

```diff
@@ -293,7 +246,7 @@ export class PolygonTool {
       gj.features.push(polygon);
     }
 
-    (this.map.getSource(source) as GeoJSONSource).setData(gj);
+    polygonToolGj.set(gj);
```

Comment:
This is the big trick I didn't think about before. We have non .svelte files doing stuff, but we want to do the rendering in normal svelte. All we have to do is plumb the GJ back to svelte using stores